draft-ietf-openpgp-pqc.md

```diff
@@ -894,19 +892,19 @@ The post-quantum KEM algorithms defined in {{kem-alg-specs}} and the signature a
 During the transition period, the post-quantum algorithms will not be supported by all clients.
 Therefore various migration considerations must be taken into account, in particular backwards compatibility to existing implementations that have not yet been updated to support the post-quantum algorithms.
 
-## Key preference
+## Key preference {#pq-key-preference}
 
 Implementations SHOULD prefer PQ(/T) keys when multiple options are available.
+For instance, an implementation that encrypts messages for a recipient for which both a valid PQ/T and a valid ECC certificate are available, the implementation should use the PQ/T certificate only.
```

Comment:
Here I think we run into the problem that the abstract concept of a recipient (outside of PKESKs) is not well defined. Might be ok for the non-normative statement, though.

Reply:
I think it's fine as an explaination of the previous sentence

Reply:
I would still rather rewrite this. IMO that problem is outside of the scope of the OpenPGP protocol and more an application thing. The protocol layer doesn't necessarily know that two certificates / keys that it processes for encrypting are somehow linked. And even if it does, it can't know the intent of the application/user.
See also #67, I think the critique still applies to this: Do we mean the second or the third scenario, or both, and what is the criterion for an application to determine that two certificates belong to the same recipient in the third scenario?

In any case, I would change the paragraph to start with the more obvious case first:

> For instance, in the case that a certificate has both a PQ/T and an ECC encryption-capable valid subkey, the PQ/T subkey SHOULD be preferred.

And then we can add the "multiple certificate scenario" a bit more vaguely. The best I could come up with is the following:

> Further, if an application can determine (by any means not defined in this document) that encrypting to a PQ/T certificate and a traditional certificate serves the same purpose, it should omit encrypting to the traditional certificate.

Sorry for being a bit picky here but I think sooner or later someone will re-raise this issue.

Reply:
After looking closer again I agree with Johannes, here the assumptions about the identity of the recipient will most likely have to be more vague.

Reply:
```suggestion
For instance, in the case that a certificate has both a PQ/T and an ECC encryption-capable valid subkey, the PQ/T subkey SHOULD be preferred.
Furthermore, if an application can anyhow determine that encrypting to a PQ/T certificate and a traditional certificate is redundant, it should omit encrypting to the traditional certificate.
```
What about this?

Reply:
I think we're almost there. But "SHOULD be preferred" might not be clear enough, as we also want to express that in this case the traditional key should not be used at all. So I would write for instance " ... valid subkey, the PQ/T subkey SHOULD be used and the traditional key SHOULD NOT be used."

Furthermore, I wouldn't write "and an *ECC* encryption-capable valid subkey" but rather "and a *traditional* encryption-capable valid subkey" in order not to create the false assumption that it only refers to ECC keys (other types of keys might be around for quite some time).

Reply:
That's the point of "for instance", IMO. I would rather remove the normative part from an example then or remove the "for instance".

Normative way:
```suggestion
In case that a certificate has both a valid PQ/T and a valid traditional subkey, an implementation SHOULD use the PQ/T subkey only.
Furthermore, if an application can anyhow determine that encrypting to a PQ/T certificate and a traditional certificate is redundant, it should omit encrypting to the traditional certificate.
```
Example way:
```suggestion
For instance, an implementation that encrypts messages for a recipient for which both a valid PQ/T and a valid ECC encryption subkey are available, the implementation should use the PQ/T subkey only.
Furthermore, if an application can anyhow determine that encrypting to a PQ/T certificate and a traditional certificate is redundant, it should omit encrypting to the traditional certificate.
```
Somehow hybrid, but not normative:
```suggestion
For instance, an implementation that encrypts messages for a recipient for which both a valid PQ/T and a valid traditional encryption subkey are available, the implementation should use the PQ/T subkey only.
Furthermore, if an application can anyhow determine that encrypting to a PQ/T certificate and a traditional certificate is redundant, it should omit encrypting to the traditional certificate.
```

Reply:
I would write:

> For instance, an implementation that encrypts a message to a certificate for which both a valid PQ/T and a valid traditional encryption subkey are available, the implementation should encrypt to the PQ/T subkey only.

This is adapted from your "Example way" to avoid the "recipient" terminology, and I also replaced "use" by "encrypt to", as well as "ECC" by "traditional".

I'm happy with your suggestion for the "multiple certificate scenario".

Reply:
I vote for the normative variant @wussler has posted above.

(An example with ECC is in my point potentially misleaing – even though from a formal point it is correct since it is an example. But our aim here by rendering the guidance as an example is only with respect to the set of keys, I don't see a need to make it concrete with respect to the public-key encryption algorithm.)

I also generally agree to the second sentence about the redundant certificate. That solves the problem of the using the word recipient well in my view. But the use of the word "anyhow" doesn't seem right to me. I ran the sentence through deepl and it doesn't translate to German correctly. I think we need to replace "anyhow" with "in some way" or with "somehow". Or maybe even better 

"Furthermore, if an application has any means to determine ..."

Reply:
Ok, then let's go with the normative variant.

I was a bit unsure about "anyhow" myself but cannot judge it. I vote for your suggestion to write "Furthermore, if an application has any means to determine ...".